First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request contains the first commits to create an Ansible role that installs and configures the Tanium client.

## 💭 Motivation and context ##

The instances in the Shared Services account must run the Tanium client and communicate with the Tanium server in the VENOM environment. 

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

jsf9k committed:
Configure TF code for this Ansible role

jsf9k committed:
Update README and metadata for this role

jsf9k committed:
Update role name

jsf9k committed:
Add tasks and make necessary adjustments to install Tanium

jsf9k committed:
Add code to configure the Tanium client

jsf9k committed:
Add test code to verify Tanium installation and configuration

jsf9k committed:
Reduce codeowners to only those familiar with this repo

jsf9k committed:
Rename repo to ansible-role-venom-tanium-client

jsf9k committed:
Pin ansible-lint since version 5 contains breaking changes

See cisagov/skeleton-ansible-role#69 for more details.

jsf9k committed:
Add role variables and dependencies to README

jsf9k committed:
Add comments explaining why Debian Bullseye and Kali are unsupported

Tanium does not yet support Debian Bullseye, and hence does not
support Kali either.

dav3r commented on terraform/user.tf:
```diff
@@ -10,8 +10,29 @@ module "user" {
 }
+
+# Attach third-party S3 bucket read-only policy to the production
+# EC2AMICreate role
```
```suggestion
# role used by the test user
```

dav3r commented on terraform/user.tf:
```diff
@@ -10,8 +10,29 @@ module "user" {
+}
+
+# Attach third-party S3 bucket read-only policy to the staging
+# EC2AMICreate role
```
```suggestion
# role used by the test user
```

dav3r reviewed (approved):
Excellent! 🍊 

I noted a couple of comments that look like they need to be updated after being copy/pasted from some other repo.

jsf9k committed:
Add support for Fedora

jsf9k committed:
Modify comments to not use the role name

The role name is a Terraform variable default, but it could be overridden by the user.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Explicitly mention Tanium client

hillaryj commented on molecule/default/tests/test_default.py:
```diff
@@ -12,7 +12,27 @@
-def test_packages(host, x):
-    """Run a dummy test, just to show what one would look like."""
-    assert x
+def test_tainum_installed(host):
```
The `i` is in the wrong place in `tanium` in this def - I'd use suggest changes, but wanted to comment instead in case the typo is carried through elsewhere.

jsf9k replied:
Great catch!  See commit 61db149.

I also did a `find . -type f -exec grep -HniF tainum {} \;` and determined that this was the only place that suffered from this particular ailment.

hillaryj reviewed (commented):
Great work!

jsf9k committed:
Correct typo in function name

Great catch by @hillaryj!

Co-authored-by: Hillary Jeffrey <hillary.jeffrey@cisa.dhs.gov>

hillaryj reviewed (approved):
🥇

mcdonnnj commented on tasks/RedHat.yml:
```diff
@@ -0,0 +1,36 @@
+---
+- name: Grab Tanium system package GPG key from S3
+  ansible.builtin.aws_s3:
```
```suggestion
  amazon.aws.aws_s3:
```

jsf9k replied:
See commit 9791169.

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,70 @@
+- name: Install Tanium
+  block:
+    - name: Grab Tanium system package from S3
+      ansible.builtin.aws_s3:
```
```suggestion
      amazon.aws.aws_s3:
```

jsf9k replied:
See commit 9791169.

mcdonnnj commented on tasks/RedHat.yml:
```diff
@@ -0,0 +1,36 @@
+    mode: 0700
+    src: /tmp/{{ package_gpg_key_object_name }}
+
+- name: Import Tanimu system package GPG key
```
```suggestion
- name: Import Tanium system package GPG key
```

jsf9k replied:
See commit 6796bd8.

mcdonnnj commented on tasks/main.yml:
```diff
@@ -1,2 +1,70 @@
+      paths:
+        - "{{ role_path }}/vars"
+
+# Check if Tanium is already installed
```
Redundant?

jsf9k replied:
See commit 9ebb35b.

mcdonnnj reviewed (commented):
Just some more FQDN and a typo. I know you're already handling the user TF module in #5. Thanks again!
